lib/jarvis/commands/publish.rb

```diff
@@ -1,23 +1,36 @@
 require "clamp"
+require "net/http"
 require "i18n"
+require "json"
 require "stud/temporary"
 require "jarvis/exec"
 require "jarvis/github/project"
+require "gems"
 
 module Jarvis module Command class Publish < Clamp::Command
+  class RemoteVersionDontMatchLocal < ::Jarvis::Error; end
+  class CommitHashDontMatch < ::Jarvis::Error; end
+  class CIBuildFail < ::Jarvis::Error; end
+  class NoGemspecFound < ::Jarvis::Error; end
+
   banner "Publish a logstash plugin"
 
-  option "--committer-email", "EMAIL", "The git committer to set on all commits. If not set, the default is whatever your git defaults to (see `git config --get user.email` and `git config --get user.email`)."
-  option "--committer-name", "NAME", "The git committer name to set on all commits. If not set, the default is whatever your git defaults to (see `git config --get user.name` and `git config --get user.email`)."
   option "--workdir", "WORKDIR", "The place where this command will download temporary files and do stuff on disk to complete a given task."
-  option "--github-token", "GITHUB_TOKEN", "Your github auth token", :required => true
+  option "--force", :flag, "Dont check if the build pass on jenkins and try to publish anyway"
```

Comment:
+1 for the naming, 

I wanted your input for that, how do we expose that function to the end user?  as a new command?